utils/filecache/cache.go

```diff
@@ -0,0 +1,200 @@
+package filecache
+
+import (
+	"context"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	fs "github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+type Cache struct {
+	entries       *entryMap
+	entriesLM     *lockMap
+	entryProvider IEntryProvider
+	fs            fs.FS
+	cfg           *FileCacheConfig
+	cancelStore   *parallelisation.CancelFunctionStore
+	closed        atomic.Bool
+}
+
+func (c *Cache) gc(ctx context.Context, _ time.Time) {
```

Comment:
it is probabaly a good idea to check the context when you loop through this since it could be a long running operation. use parallelisation.DetermineContextError (or whatever it is called) so you can break early if the context is cancelled

Reply:
also there should be a test that this works